sw/device/silicon_creator/lib/manifest_def.c

```diff
@@ -0,0 +1,20 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <stdint.h>
+
+#include "sw/device/silicon_creator/lib/manifest_def.h"
+
+/*
+ * Declarations for the manifest fields populdated by the linker script.
+ */
+extern const uint32_t _manifest_code_start;
+extern const uint32_t _manifest_code_end;
+extern const uint32_t _manifest_entry_point;
+
+const volatile manifest_t kManifest = {
```

Comment:
Why's this volatile?

Reply:
Only a few of the fields are populated in the struct definition, the remainder are added to the binary by the signer tool. The volatile here is to prevent future reads on those fields from getting optimized out. I'm open to better ideas if you have any, otherwise I can add a comment about this.

Reply:
Can you include a comment specifying that this is happening? Otherwise yes, this is (unfortunately) correct.